wfl/fit/ace.py

```diff
@@ -138,49 +228,63 @@ def _check_output_files(file_base, formats):
 
         return results
 
-    if not run_dir.exists():
-        run_dir.mkdir(parents=True)
+    Path(run_dir).mkdir(exist_ok=True, parents=True)
 
-    use_params = dict(params)
+    use_params = deepcopy(ace_fit_params)
 
-    if dry_run:
-        use_params['dry_run'] = None
+    _write_fitting_configs(fitting_configs, use_params, ace_file_base)
 
-    # the code below needs to know an unfortunate amount about the inner workings of ace_fit.jl
-    # here we rely on knowledge of the --outfile_base
-    use_params['outfile_base'] = str(ace_file_base)
-    use_params['outfile_format'] = formats
+    ace_fit_params_filename = ace_file_base + "_fit_params.json"
+    with open(ace_fit_params_filename, "w") as f:
+        f.write(json.dumps(use_params, indent=4))
 
-    use_params['key'] = [ [k, ref_property_prefix + v] for k,v in [('E', 'energy'), ('F', 'forces'), ('V', 'virial')] ]
-    use_params['_repeat_key'] = True
+    cmd = f"{ace_fit_exec} --fit-params {ace_fit_params_filename} "
+    if dry_run: 
+        cmd += "--dry-run "
+    cmd +=  f"> {ace_file_base}.stdout 2> {ace_file_base}.stderr "
 
-    # configs need to be in memory so they can be modified with stress -> virial, and safest to
-    # have them as a list (rather than using ConfigSet_in.to_memory()) when passing to ase.io.write below
-    fitting_configs = list(fitting_configs)
+    if verbose:
+        print('fitting command:\n', cmd)
 
-    # calculate virial from stress, since ASE uses stress but ace_fit.jl only knows about virial
-    for at in fitting_configs:
-        if ref_property_prefix + 'stress' in at.info:
-            stress = at.info[ref_property_prefix + 'stress']
-            if len(stress.shape) == 2:
-                # 3x3
-                at.info[ref_property_prefix + 'virial'] = list((-stress * at.get_volume()).ravel())
-            else:
-                # Voigt 6-vector
-                at.info[ref_property_prefix + 'virial'] = list((-voigt_6_to_full_3x3_stress(stress) * at.get_volume()).ravel())
+    return _execute_fit_command(cmd, ace_file_base, dry_run)
 
-    # write to file since that's what script needs as input
-    fitting_configs_filename = str(run_dir / f'fitting_database.{ACE_name}.extxyz')
-    ase.io.write(fitting_configs_filename, fitting_configs)
 
-    use_params["atoms_filename"] = fitting_configs_filename
+def _write_fitting_configs(fitting_configs, use_params, ace_file_base):
+    """
+    Writes fitting configs to file and updates ace fitting parameters. 
+    Configurations and filename handled by Workflow overwrite any filename
+    specified in parameters. 
+    """
 
-    fitting_line = dict_to_ace_fit_string(use_params)
+    if "data" not in use_params.keys():
+        use_params["data"] = {}
 
-    cmd = f'{ace_fit_exec} {fitting_line} > {ace_file_base}.stdout 2> {ace_file_base}.stderr'
+    fit_cfgs_fname_given = None 
+    if "fname" in use_params["data"].keys():
```

Comment:
No need for the `.keys()` part - `in dict` is enough.  Also, maybe cleaner with `fit_cfgs_fname_given = use_params["data"].get("fname", None)`?

Reply:
Doing this as part of my final (?) round of changes.